skore/tests/unit/displays/table_report/test_cross_validation.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.cluster import KMeans
+from sklearn.datasets import make_regression
+from skore import CrossValidationReport, Display, TableReportDisplay
+from skore._externals._skrub_compat import tabular_pipeline
+
+
+@pytest.fixture
+def cross_validation_report():
+    X, y = make_regression(n_samples=100, n_features=5, random_state=42)
+    X = pd.DataFrame(X, columns=[f"Feature_{i}" for i in range(5)])
+    y = pd.Series(y, name="Target_")
+    return CrossValidationReport(tabular_pipeline("regressor"), X=X, y=y)
+
+
+@pytest.fixture
+def display(cross_validation_report):
+    return cross_validation_report.data.analyze()
+
+
+def test_table_report_display_constructor(display):
+    """Check the value that are stored in the display constructor."""
+    assert isinstance(display, Display)
+
+    assert hasattr(display, "summary")
+    assert isinstance(display.summary, dict)
+    assert set(display.summary.keys()).issuperset(
+        {
+            "dataframe",
+            "dataframe_module",
+            "n_rows",
+            "n_columns",
+            "columns",
+            "dataframe_is_empty",
+            "plots_skipped",
+            # skrub>=0.6
+            # "associations_skipped",
+            "sample_table",
+            "n_constant_columns",
+            "top_associations",
+        }
+    )
+
+
+def test_table_report_display_frame(cross_validation_report):
+    display = cross_validation_report.data.analyze()
+    dataset = display.frame(kind="dataset")
+
+    pd.testing.assert_frame_equal(
+        dataset,
+        pd.concat([cross_validation_report.X, cross_validation_report.y], axis=1),
+    )
+
+    associations = display.frame(kind="top-associations")
+    pd.testing.assert_frame_equal(
+        associations, pd.DataFrame(display.summary["top_associations"])
+    )
+
+
+@pytest.mark.parametrize(
+    "X",
+    [
+        np.random.rand(100, 5),
+        pd.DataFrame(
+            np.random.rand(100, 5), columns=[f"Feature number {i}" for i in range(5)]
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "y",
+    [
+        np.ones((100, 1)),
+        np.ones(100),
+        # pd.Series(np.ones(100)),
+        # pd.DataFrame(np.ones((100, 1)), columns=["Target"]),
+    ],
+)
+def test_retrieve_data_as_frame(X, y):
```

Comment:
It is a bit weird for me to call this function `retrieve_data_as_frame` while we don't have an assertion in this direction.
I would suggest to make some assertion the retrieved dataframe store in `display.summary["dataframe"]` that should be concatenation of the `X` and `y` that we have in the parametrization. I think that the test does not need to check up to the data in the dataframe but at least the shape and the column names.

Reply:
Also it would be nice to have a small docstring explaining what we try to to check for the function.